### PR TITLE
[refactor](bitmap)  using template to reduce duplicate code

### PR DESCRIPTION
**Pull request by mrhhsg**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by mrhhsg** [comment]
run buildall

**Review by yiguolei — DISMISSED** [review]
LGTM

File: be/src/util/bitmap_value.h [hunk 1]
```diff
@@ -187,16 +187,25 @@ class Roaring64Map {
...
-            roarings[0].add(vals[lcv]);
-            roarings[0].setCopyOnWrite(copyOnWrite);
+    template <typename T>
+    void addMany(size_t n_args, const T* vals) {
+        if constexpr (sizeof(T) == 4) {
+            auto& roaring = roarings[0];
+            roaring.addMany(n_args, reinterpret_cast<const uint32_t*>(vals));
+            roaring.setCopyOnWrite(copyOnWrite);
+        } else if constexpr (sizeof(T) < 8) {
+            auto& roaring = roarings[0];
```
Comment (yiguolei) [inline be/src/util/bitmap_value.h:197]: is this branch meaningful?

File: be/src/util/bitmap_value.h [hunk 2]
```diff
@@ -1231,6 +1240,49 @@ class BitmapValue {
         }
     }
 
+    template <typename T>
+    void add_many(const T* values, const size_t count) {
+        switch (_type) {
+        case EMPTY:
+            if (config::enable_set_in_bitmap_value && count < SET_TYPE_THRESHOLD) {
+                for (size_t i = 0; i != count; ++i) {
```
Comment (yiguolei) [inline be/src/util/bitmap_value.h:1248]: 这个如果count == 1 怎么办？

**Comment by mrhhsg** [comment]
run buildall

**Comment by mrhhsg** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.4 seconds
 stream load tsv:          541 seconds loaded 74807831229 Bytes, about 131 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.7 seconds inserted 10000000 Rows, about 336K ops/s
 storage size: 17162313057 Bytes

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.59 seconds
 stream load tsv:          541 seconds loaded 74807831229 Bytes, about 131 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17162310174 Bytes

**Review by yiguolei — DISMISSED** [review]
LGTM

**Review by jacktengg — APPROVED** [review]
LGTM

File: be/src/util/bitmap_value.h [hunk 3]
```diff
@@ -187,16 +187,25 @@ class Roaring64Map {
...
-            roarings[0].setCopyOnWrite(copyOnWrite);
+    template <typename T>
+    void addMany(size_t n_args, const T* vals) {
+        if constexpr (sizeof(T) == sizeof(uint32_t)) {
+            auto& roaring = roarings[0];
+            roaring.addMany(n_args, reinterpret_cast<const uint32_t*>(vals));
+            roaring.setCopyOnWrite(copyOnWrite);
+        } else if constexpr (sizeof(T) < sizeof(uint32_t)) {
+            auto& roaring = roarings[0];
+            std::vector<uint32_t> values(n_args);
```
Comment (jacktengg) [inline be/src/util/bitmap_value.h:198]: values no filled?

**Comment by mrhhsg** [comment]
run buildall

**Review by jacktengg — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.67 seconds
 stream load tsv:          541 seconds loaded 74807831229 Bytes, about 131 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17161963357 Bytes

**Pull request merged**